python/doc/examples/numerical_methods/general_methods/plot_estimate_integral_iterated_quadrature.py

```diff
@@ -1,48 +1,272 @@
+g.setYTitle('y')
+
+# %%
+# We compute the integral using an  :class:`~openturns.IteratedQuadrature`
+# algorithm with
+# the :class:`~openturns.GaussKronrodRule`  *G11K23*, which means that we use
+# on each subinterval:
+#
+# - 11 points for the Gauss approximations,
+# - 23 points for the Kronrod approximation including the 11 previous ones.
+#
+# We limit the number of subintervals and we  define a maximum error between both approximations:
+# when the difference between both approximations is lower, a new subdivision is performed.
+maximumSubIntervals = 32
+maximumError = 1e-4
+GKRule = ot.GaussKronrodRule(ot.GaussKronrodRule.G1K3)
+integration_algo = ot.IteratedQuadrature(ot.GaussKronrod(maximumSubIntervals, maximumError, GKRule))
+
+# %%
+# To get the nodes used to approximate the integral, we need to use a :class:`~openturns.MemoizeFunction`
+# that stores all the calls to the function.
+integrand_memoized = ot.MemoizeFunction(integrand)
+I_value = integration_algo.integrate(integrand_memoized, domain)
+print('I = ', I_value[0])
+nodes = integrand_memoized.getInputHistory()
+print('Nodes : ', nodes)
+
+# %%
+# We can draw in black the nodes on the graph that contains the iso-lines of the
+# function :math:`f`. We can see that the algorithm focuses on the nodes where the
+# function has its discontinuities.
+cloud_nodes = ot.Cloud(nodes)
+cloud_nodes.setPointStyle('dot')
+cloud_nodes.setColor('blue')
+
+g.add(cloud_nodes)
+g.setLegendPosition('')
+view = otv.View(g)
+
+# %%
+# Case 2: The integrand function has large gradients.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We consider the function:
+#
+# .. math::
+#
+#    f : \begin{array}{lcl}
+#          \Rset^2 & \rightarrow & \Rset \\
+#          (x,y) & \rightarrow & e^{-(x^2+y^2)} + e^{-8((x-4)^2+(y-4)^2)}
+#          \end{array}
+#
+# the domain :math:`\set{D} =[-2,2]\times [-2,2]` and the integral:
+#
+# .. math::
+#    I = \int_{\set{D}} f(x,y)\,dxdy.
+
+# %%
+# We first define the integrand :math:`f` and the domain :math:`\set{D}`.
+
+
+def kernel2(x):
+    return [m.exp(-(x[0]**2 + x[1]**2)) + m.exp(-8 * ((x[0] - 4)**2 + (x[1] - 4)**2))]
+
+
+integrand = ot.PythonFunction(2, 1, kernel2)
+domain = ot.Interval([-2.0, -2.0], [6.0, 6.0])
 
 # %%
-# define the integrand and the bounds
-a = -m.pi
-b = m.pi
-f = ot.SymbolicFunction(["x", "y"], ["1+cos(x)*sin(y)"])
-ll = [ot.SymbolicFunction(["x"], [" 2+cos(x)"])]
-u = [ot.SymbolicFunction(["x"], ["-2-cos(x)"])]
+# We draw the iso-lines of the integrand function.
+g = integrand.draw([-3.0, -3.0], [7.0, 7.0])
+g.setTitle(r'Function $f(x,y) = e^{-(x^2+y^2)} + e^{-8((x-4)^2+(y-4)^2)} $')
+g.setXTitle('x')
+g.setYTitle('y')
+
+# %%
+# To get the nodes used to approximate the integral, we need to use a :class:`~openturns.MemoizeFunction`
+# that stores all the calls to the function.
+integrand_memoized = ot.MemoizeFunction(integrand)
+I_value = integration_algo.integrate(integrand_memoized, domain)
+print('I = ', I_value[0])
```

Comment:
The exact value is 3.14838390938152462959575371724 according to Maple

Reply:
ok